x/lockup/keeper/lock.go

```diff
@@ -525,6 +560,10 @@ func (k Keeper) ForceUnlock(ctx sdk.Context, lock types.PeriodLock) error {
 	return k.unlockInternalLogic(ctx, lock)
 }
 
+func (k Keeper) BreakLockForUnpool(ctx sdk.Context, lock types.PeriodLock) error {
+	return k.unlockInternalLogic(ctx, lock)
```

Comment:
UnlockInternalLogic assumes the lock is unlocking(see KeyPrefixUnlocking usage). This is not always true for unpooling.

Reply:
good catch!